.github/workflows/bench.yml

```diff
@@ -1,52 +1,159 @@
 name: "Run benchmark"
-
 on:
   pull_request_target:
     types: [assigned, opened, synchronize, reopened, edited]
   push:
     branches:
-      - main
-
+      - benchmarks_in_parallel
 permissions:
   contents: write
   pull-requests: write
   issues: write
-
 jobs:
-  build:
+  benchmark:
     runs-on: benchmarking-runner
     if: github.event.head_commit.message != 'Update performance results in README.md'
     env:
       GITHUB_TOKEN: ${{ secrets.GITHUB_TOKEN }}
+    strategy:
+      matrix:
+        service: [apollo_server, caliban, netflix_dgs, gqlgen, tailcall, async_graphql, hasura, graphql_jit]
     steps:
       - name: Checkout (GitHub)
         uses: actions/checkout@v4
         with:
           token: ${{ secrets.GITHUB_TOKEN }}
           ref: ${{ github.event_name == 'pull_request_target' && github.event.pull_request.head.sha || github.sha }}
-
-      - name: Build devcontainer and run benchmarks
+      
+      - name: Build devcontainer and run setup and benchmark
         uses: devcontainers/ci@v0.3
         with:
           imageName: graphql-benchmarks
           push: never
           runCmd: |
+            echo "Running setup script"
             bash ./setup.sh
-            bash ./run_benchmarks.sh
+            echo "Setup completed"
+            echo "Running benchmark for ${{ matrix.service }}"
+            chmod +x ./run_benchmarks.sh
+            valid_services=("apollo_server" "caliban" "netflix_dgs" "gqlgen" "tailcall" "async_graphql" "hasura" "graphql_jit")
+            if [[ ! " ${valid_services[@]} " =~ " ${{ matrix.service }} " ]]; then
+                echo "Error: Invalid service name '${{ matrix.service }}'"
+                exit 1
+            fi
+            if ! ./run_benchmarks.sh ${{ matrix.service }}; then
+                echo "Error: Benchmark failed for ${{ matrix.service }}"
+                exit 1
+            fi
+            echo "Benchmark completed for ${{ matrix.service }}"
```

Comment:
May be we need to create a `setup-{matrix}.sh` file so that each server can set this up independently.

Reply:
Am not sure i get this part..

Reply:
We are calling `setup` everytime for all graphql servers. Instead we should setup only the one that we want to run.

Reply:
okay

Reply:
The command should be moved to a script file so that it can be called directly from the CI. We don't want to touch CI file everytime we make a change.

Reply:
We should ideally have three scripts — 
1. `{matrix.service}/setup.sh`
2. `{matrix.service}/start.sh`
3. `{matrix.service}/test.sh`
4. `{matrix.service}/stop.sh` 

And from the CI we should run these scripts in separate steps.

Reply:
Ensure the `setup` scripts install everything and start the server.